addons/calendar/models/res_partner.py

```diff
@@ -39,18 +40,18 @@ def _compute_meeting(self):
             meeting_data = self.env.cr.fetchall()
 
             # Create a dict {partner_id: event_ids} and fill with events linked to the partner
-            meetings = {p["id"]: set() for p in all_partners}
-            for m in meeting_data:
-                meetings[m[0]].add(m[1])
+            meetings = {}
+            for p_id, m_id, _ in meeting_data:
+                meetings.setdefault(p_id, set()).add(m_id)
 
             # Add the events linked to the children of the partner
-            for p in all_partners:
-                partner = p
-                while partner:
-                    if partner["id"] in self.ids:
-                        meetings[partner["id"]] |= meetings[p["id"]]
-                    partner = next((pt for pt in all_partners if partner["parent_id"] and pt["id"] == partner["parent_id"][0]), None)
-            return {p_id: list(meetings[p_id]) for p_id in self.ids}
+            for meeting_partner in set(meetings):
```

Comment:
First I put `meetings.keys()` but one ci test did not pass because it's the equivalent of directly looping through `meetings`. So I did that but it crashed because we are editing meetings while looping through it. So i found out that `set(meetings)` was the only way..

Reply:
That's indeed not a good idea to loop on a dict you are modifying. Can't you build a final (independent) dict instead ?

Reply:
Well that's basically the point of creating a new set on which we loop through :thinking:. `meetings` is thus already the "independent" dict.